platform/android/export/export.cpp

```diff
@@ -1341,6 +1356,40 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 			return str;
 		}
 	}
+
+	// Creates strings.xml files inside the gradle project for different locales.
+	void _create_project_name_strings_files(const Ref<EditorExportPreset> &p_preset) {
+		const char *locales[] = {
+			"ar", "bg", "ca", "cs", "da", "de", "el", "en", "es", "fa", "fi", "fr", "hi", "hr",
+			"hu", "in", "it", "iw", "ja", "ko", "lt", "lv", "nb", "nl", "pl", "pt", "ro", "ru",
+			"sk", "sl", "sr", "sv", "th", "tl", "tr", "uk", "vi", "zh", "zh_HK", "zh_TW", "es-ES"
+		};
+		// TODO: use package/name or application/config/name?
+		String package_name = p_preset->get("package/name");
+		String project_name = get_project_name(package_name);
+		size_t num_locales = sizeof(locales) / sizeof(locales[0]);
+
+		// Creates and stores the default strings.xml file.
+		const String default_xml_string = "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n"
+					  "<resources>\n\t<string name=\"godot_project_name_string\">" + project_name + "</string>\n</resources>";
+		store_string_at_path("res://android/build/res/values/strings.xml", default_xml_string);
+
+		for (int i = 0; i < num_locales; i++) {
+			String lang = locales[i];
+			String prop = "application/config/name_" + lang;
+			String hyphen_lang = lang.replace("_", "-r");
+			String locale_directory = "res://android/build/res/values-" + hyphen_lang + "/strings.xml";
+			if (ProjectSettings::get_singleton()->has_setting(prop)) {
+				String locale_project_name = ProjectSettings::get_singleton()->get(prop);
+				String xml_string = "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n"
+				 "<resources>\n\t<string name=\"godot_project_name_string\">" + locale_project_name + "</string>\n</resources>";
+				store_string_at_path(locale_directory, xml_string);
+			} else {
+				store_string_at_path(locale_directory, default_xml_string);
```

Comment:
Add a TODO to remove this line if we remove the placeholder files from the default gradle project.